### DISCUSSION TITLE
Regarding move task from one VM to another

### DISCUSSION THREAD

**Discussion by HilmyThoriq** [manoelcampos/cloudsim-plus#441] [Q&A] [answered]
Hello.
I want to create a simulation with load balancer concept, since load balancer is a vm i want to make the task is move first to the load balancer and then after some calculation for task distribution it will move the task from load balancer to the assigned vm?
My question is can i do it? 
or maybe can you provide the example of moving task from one vm to another but if i move it i don't need to pause the simulation

**Comment by manoelcampos** [comment, marked as answer]
If what you are trying to do Cloudlet migration, despite there was some methods such as migrateCloudlet in the Datacenter class (inherited from CloudSim), this feature is not working in CloudSim or CloudSim Plus.

In fact, this is not a feature that I intend to implement since the migration of applications doesn't make much sense.
In real scenarios, it is difficult to migrate an app from a VM to another, since you have to carry all the configuration and make sure that software dependencies will be installed and all the environment configured again.

Therefore, app migration is an impractical mechanism. If you need to migrate something, you have to migrate the VM from a Host to another. If the application is running inside some container such as Docker, then you migrate the whole container.
That is because the container encapsulates all configurations and dependencies. 

CloudSim Plus doesn't have container support and the way this feature was implemented in CloudSim was the reason for me to start CloudSim Plus as an independent and re-engineered project. I wish to implement containers in CloudSim Plus but I don't have a schedule yet to provide this feature.
